sdk/servicebus/azure-servicebus/tests/test_mgmt_queues.py

```diff
@@ -0,0 +1,198 @@
+from azure.servicebus._common.utils import utc_now
+from azure.core.exceptions import ResourceExistsError, HttpResponseError
+from devtools_testutils import AzureMgmtTestCase, CachedResourceGroupPreparer
+from servicebus_preparer import CachedServiceBusNamespacePreparer, ServiceBusQueuePreparer, CachedServiceBusQueuePreparer
+from utilities import get_logger
+
+_logger = get_logger(logging.DEBUG)
+
+class ServiceBusMgmtQueueTests(AzureMgmtTestCase):
+
+    @pytest.mark.liveTest
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    def test_mgmt_queue_create_by_name(self, servicebus_namespace_connection_string, **kwargs):
+        mgmt_service = ServiceBusManagementClient.from_connection_string(servicebus_namespace_connection_string)
+
+        queue_name = str(uuid.uuid4())
+        created_at = utc_now()
+        mgmt_service.create_queue(queue_name)
+
+        queue = mgmt_service.get_queue(queue_name)
+        assert queue.queue_name == queue_name
+        assert queue.entity_availability_status == 'Available'
+        assert queue.status == 'Available'
+        assert created_at < queue.created_at < utc_now() + datetime.timedelta(minutes=10) # TODO: Should be created_at_utc for consistency with dataplane.
+
+    @pytest.mark.liveTest
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    def test_mgmt_queue_create_with_invalid_name(self, servicebus_namespace_connection_string, **kwargs):
+        mgmt_service = ServiceBusManagementClient.from_connection_string(servicebus_namespace_connection_string)
+
+        with pytest.raises(TypeError):
+            mgmt_service.create_queue(Exception())
+
+        with pytest.raises(ValueError):
+            mgmt_service.create_queue(QueueDescription(queue_name=Exception()))
+
+        with pytest.raises(ValueError):
+            mgmt_service.create_queue('')
+
+        with pytest.raises(ValueError):
+            mgmt_service.create_queue(QueueDescription(queue_name=''))
+
+    @pytest.mark.liveTest
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    def test_mgmt_queue_create_with_queue_description(self, servicebus_namespace_connection_string, **kwargs):
+        mgmt_service = ServiceBusManagementClient.from_connection_string(servicebus_namespace_connection_string)
+
+        queue_name = str(uuid.uuid4())
+        #TODO: Why don't we have an input model (queueOptions? as superclass of QueueDescription?) and output model to not show these params?
+        #TODO: This fails with the following: E           msrest.exceptions.DeserializationError: Find several XML 'prefix:DeadLetteringOnMessageExpiration' where it was not expected .tox\whl\lib\site-packages\msrest\serialization.py:1262: DeserializationError
```

Comment:
Can nix this 2nd todo if your fixes addressed it.

as to the former TODO, I notice Anna mentioned something similar higher up in the PR, if this is going to be fixed now we can remove then, otherwise should probably put a backlog issue so we remember this and yank the line here for better tracking.

Reply:
I'll create the convenience layer model today. Not many to create.

Reply:
Second TODO has been fixed.
For the first TODO, I've created a convenience layer model. But I haven't specify which propertes to be read-only. It's actually controversial to make them read-only. This needs discussion with more people and other languages.